src/python/grpcio_observability/grpc_observability/client_call_tracer.cc

```diff
@@ -0,0 +1,242 @@
+    const grpc_core::SliceBuffer& /*send_message*/) {
+  ++sent_message_count_;
+}
+
+
+void PythonOpenCensusCallTracer::PythonOpenCensusCallAttemptTracer::RecordReceivedMessage(
+    const grpc_core::SliceBuffer& /*recv_message*/) {
+  ++recv_message_count_;
+}
+
+
+namespace {
+
+void FilterTrailingMetadata(grpc_metadata_batch* b, uint64_t* elapsed_time) {
+  if (PythonOpenCensusStatsEnabled()) {
+    absl::optional<grpc_core::Slice> grpc_server_stats_bin =
+        b->Take(grpc_core::GrpcServerStatsBinMetadata());
+    if (grpc_server_stats_bin.has_value()) {
+      ServerStatsDeserialize(
+          reinterpret_cast<const char*>(grpc_server_stats_bin->data()),
+          grpc_server_stats_bin->size(), elapsed_time);
+    }
+  }
+}
+
+}  // namespace
+
+
+void PythonOpenCensusCallTracer::PythonOpenCensusCallAttemptTracer::
+    RecordReceivedTrailingMetadata(
+        absl::Status status, grpc_metadata_batch* recv_trailing_metadata,
+        const grpc_transport_stream_stats* transport_stream_stats) {
+  status_code_ = status.code();
+  if (PythonOpenCensusStatsEnabled()) {
+    uint64_t elapsed_time = 0;
+    if (recv_trailing_metadata != nullptr) {
+      FilterTrailingMetadata(recv_trailing_metadata, &elapsed_time);
+    }
+
+    std::string final_status = absl::StatusCodeToString(status_code_);
+    std::vector<Label> labels = context_.Labels();
+    labels.emplace_back(Label{kClientMethod, std::string(parent_->method_)});
+    labels.emplace_back(Label{kClientStatus, final_status});
+    RecordDoubleMetric(kRpcClientSentBytesPerRpcMeasureName, static_cast<double>(transport_stream_stats != nullptr ? transport_stream_stats->outgoing.data_bytes : 0), labels);
+    RecordDoubleMetric(kRpcClientReceivedBytesPerRpcMeasureName, static_cast<double>(transport_stream_stats != nullptr ? transport_stream_stats->incoming.data_bytes : 0), labels);
+    RecordDoubleMetric(kRpcClientServerLatencyMeasureName, absl::ToDoubleMilliseconds(absl::Nanoseconds(elapsed_time)), labels);
+    RecordDoubleMetric(kRpcClientRoundtripLatencyMeasureName, absl::ToDoubleMilliseconds(absl::Now() - start_time_), labels);
+    if (grpc_core::IsTransportSuppliesClientLatencyEnabled()) {
+      if (transport_stream_stats != nullptr && gpr_time_cmp(transport_stream_stats->latency,
+                        gpr_inf_future(GPR_TIMESPAN)) != 0) {
+        double latency_ms = absl::ToDoubleMilliseconds(absl::Microseconds(
+            gpr_timespec_to_micros(transport_stream_stats->latency)));
+        RecordDoubleMetric(kRpcClientTransportLatencyMeasureName, latency_ms, labels);
+      }
+    }
+  }
+}
+
+
+void PythonOpenCensusCallTracer::PythonOpenCensusCallAttemptTracer::RecordCancel(
+    absl::Status /*cancel_error*/) {}
+
+
+void PythonOpenCensusCallTracer::PythonOpenCensusCallAttemptTracer::RecordEnd(
+    const gpr_timespec& /*latency*/) {
+  if (PythonOpenCensusStatsEnabled()) {
+    std::vector<Label> labels = context_.Labels();
```

Comment:
Instead of copying to a local variable, can you just use `context_.Labels()` in this function where you're currently using `labels`? You're about to delete the context anyway, so it seems like adding to the labels shouldn't hurt anything.